markets/perps-market/contracts/storage/PerpsMarket.sol

```diff
@@ -141,8 +143,20 @@ library PerpsMarket {
         Position.Data memory newPosition
     ) internal returns (MarketUpdateData memory) {
         Position.Data storage oldPosition = self.positions[accountId];
+
         int128 oldPositionSize = oldPosition.size;
 
+        // update the market debt correction accumulator before losing oldPosition details
+        // by adding the new updated notional (old - new size) plus old position pnl
+        // debtCorrectionAccumulator += newPosition.size * currentPrice.toInt() - oldPosition.size * currentPrice.toInt() + oldPositionPnl ;
+        uint currentPrice = newPosition.latestInteractionPrice;
+        (int oldPositionPnl, , , ) = oldPosition.getPnl(currentPrice);
+
+        self.debtCorrectionAccumulator +=
+            currentPrice.toInt() *
+            (newPosition.size - oldPositionSize) -
```

Comment:
looks like this is `+` oldPositionPnl in the sheet?  i could be wrong but just double check